core-plugins/src/main/java/co/cask/hydrator/plugin/common/KinesisUtil.java

```diff
@@ -0,0 +1,150 @@
+   */
+  public static void createAndWaitForStream(AmazonKinesisClient kinesisClient, String streamName, int shardCount) {
+    if (streamExists(kinesisClient, streamName)) {
+      StreamStatus streamStatus = getStreamState(kinesisClient, streamName);
+      switch (streamStatus) {
+        case DELETING:
+          long waitTimeDelete = System.currentTimeMillis() + TIMEOUT;
+          while (System.currentTimeMillis() < waitTimeDelete && streamExists(kinesisClient, streamName)) {
+            try {
+              LOG.debug("Deleting Stream {} ", streamName);
+              TimeUnit.MILLISECONDS.sleep(SLEEP_INTERVAL);
+            } catch (InterruptedException e) {
+              Thread.currentThread().interrupt();
+            }
+          }
+          if (streamExists(kinesisClient, streamName)) {
+            throw new IllegalStateException(String.format("Timed out waiting for stream {} to delete", streamName));
+          }
+          createStream(streamName, shardCount, kinesisClient);
+          break;
+        case ACTIVE:
+          LOG.info("Stream {} already exists", streamName);
+          return;
+        case CREATING:
+          LOG.info("Stream {} is being created", streamName);
+          break;
+        case UPDATING:
+          LOG.info("Stream {} is being updated", streamName);
+          return;
+        default:
+          throw new IllegalStateException(String.format("Illegal stream state: {}", streamStatus));
+      }
+    } else {
+      createStream(streamName, shardCount, kinesisClient);
+    }
+    long waitTimeCreate = System.currentTimeMillis() + TIMEOUT;
+    while (System.currentTimeMillis() < waitTimeCreate) {
+      try {
+        TimeUnit.MILLISECONDS.sleep(SLEEP_INTERVAL);
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+      }
+      try {
+        StreamStatus streamStatus = getStreamState(kinesisClient, streamName);
+        if (StreamStatus.ACTIVE == streamStatus) {
+          LOG.info("Stream {} became active", streamName);
+          return;
+        }
+      } catch (ResourceNotFoundException e) {
+        throw new IllegalStateException(String.format("Stream %s did not become active in %d",
+                                                      streamName, TIMEOUT), e);
+      }
+    }
+    if (!(getStreamState(kinesisClient, streamName) == StreamStatus.ACTIVE)) {
+      throw new IllegalStateException(String.format("Stream %s did not become active in %d",
+                                                    streamName, TIMEOUT));
+    }
+  }
+
+  private static void createStream(String streamName, int shardCount, AmazonKinesisClient kinesisClient) {
+    CreateStreamRequest createStreamRequest = new CreateStreamRequest();
+    createStreamRequest.setStreamName(streamName);
+    createStreamRequest.setShardCount(shardCount);
+    kinesisClient.createStream(createStreamRequest);
+    LOG.info("Stream {} is being created", streamName);
```

Comment:
createStream method just initializes the creation. Actual creation happens after a while